### PR TITLE
Deprecate DEVICE_INIT

### PR DESCRIPTION
**Pull request by galak**
There are only a small handful of DEVICE_INIT users so lets just use DEVICE_DEFINE/SYS_DEVICE_DEFINE in those places and reduce one macro.

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]
and next: deprecating DEVICE_AND_API_INIT ;)

**Comment by galak** [comment]
> and next: deprecating DEVICE_AND_API_INIT ;)

Yeah, figured that one we can do after #29644 gets merged in and we convert most users to `DEVICE_DT_DEFINE` :)

**Review by erwango — APPROVED** [review]

File: samples/bluetooth/hci_uart/src/main.c
Comment (joerchan) [inline samples/bluetooth/hci_uart/src/main.c:326]: What is the difference between DEVICE_DEFINE and SYS_DEVICE_DEFINE and why would it be preffered to use the SYS_ versios in this case? 

I'm just curious. Is it something magic with how Z_SYS_NAME?

**Review by MaureenHelm — APPROVED** [review]

File: samples/bluetooth/hci_uart/src/main.c
Comment (galak) [reply samples/bluetooth/hci_uart/src/main.c]: I converted this to SYS_DEVICE_DEFINE as it seemed like the only reason its here is to get `hci_uart_init` called and the lack of any data pointers leaned me in that way.  @tbursztyka can probably give a better understanding of the difference between SYS_DEVICE_DEFINE and DEVICE_DEFINE usage.

**Review by joerchan — APPROVED** [review]

File: samples/bluetooth/hci_uart/src/main.c
Comment (nashif) [reply samples/bluetooth/hci_uart/src/main.c]: was wondering about this as well, this should be another candidate for deprecation or renaming, add SYS_ as a prefix does not really say anything about what is actually does.

Comment (galak) [reply samples/bluetooth/hci_uart/src/main.c]: I'll let @tbursztyka comment on that since we added SYS_DEVICE_DEFINE recently:

commit 8d7bb8ffd8faba73628e18e4651e20896da9751e
Author: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
Date:   Mon Mar 9 11:02:20 2020 +0100

    device: Refactor device structures

**Review by tyrantlizard77 — APPROVED** [review]

File: include/device.h
Comment (mniestroj) [inline include/device.h:418]: I think it would be better to get rid of `device_pm_control_nop()` function, replace `device_pm_control_nop` use with `NULL` and simply return `-ENOTSUP` here.

File: subsys/power/device.c
Comment (mniestroj) [inline subsys/power/device.c:164]: If we would replace all `device_pm_control_nop` with `NULL`, then there is no need to check for two cases here.

File: include/device.h
Comment (pabigot) [inline include/device.h:386]: I'd like to see this as:
```
typedef int (*device_pm_control_fn)(const struct device *dev, uint32_t command,
                                    void *context, device_pm_cb cb, void *arg);

...

   device_pm_control_fn pmc = dev->device_pm_control;

   if (pmc == NULL) {
      pmc = device_pm_control_nop;
   }

   return pcm(args...);
```

to reduce code size by getting rid of the replicated call prep.  But that can be done when we clean up this API for device power management.

File: include/device.h
Comment (pabigot) [reply include/device.h:418]: Possibly, but this all has to be revisited anyway.  I remember at one point there were problems with the power management functions returned an error, which is part of why the nop implementation exists.

**Review by pabigot — APPROVED** [review]

File: samples/bluetooth/hci_uart/src/main.c
Comment (tbursztyka) [reply samples/bluetooth/hci_uart/src/main.c]: SYS_DEVICE_DEFINE is for non subsystem devices (spi, i2c, adc, ...), so no data/config nor api, that may expose PM stuff as well.

Maybe we could get rid of it as well, and just use DEVICE_DEFINE. I have no strong opinion on that one.

**Pull request merged**
